packages/react-components/eslint-plugin-react-components/src/rules/prefer-fluentui-v9.ts

```diff
@@ -0,0 +1,192 @@
+                      fluent9: COMPAT_MIGRATIONS[name].component,
+                      package: COMPAT_MIGRATIONS[name].package,
+                    },
+                  });
+                } else if (isPreviewMigration(name) && unstable) {
+                  context.report({
+                    node,
+                    messageId: 'replaceFluent8With9Unstable',
+                    data: {
+                      fluent8: name,
+                      fluent9: PREVIEW_MIGRATIONS[name].component,
+                      package: PREVIEW_MIGRATIONS[name].package,
+                    },
+                  });
+                }
+                break;
+            }
+          }
+        }
+      },
+    };
+  },
+});
+
+type Migration = string | { component: string; package: string };
+
+/**
+ * Migrations from Fluent 8 components to Fluent 9 components as of Fluent UI 9.46.3
+ * @see https://react.fluentui.dev/?path=/docs/concepts-migration-from-v8-component-mapping--page
+ */
+const MIGRATIONS = {
+  mergeStyles: 'makeStyles',
+  ActionButton: 'Button',
+  Announced: 'useAnnounce',
+  Breadcrumb: 'Breadcrumb',
+  CommandBar: 'Toolbar',
+  CommandBarButton: 'ToolbarButton',
+  CommandButton: 'MenuButton',
+  CompoundButton: 'CompoundButton',
+  Callout: 'Popover',
+  Checkbox: 'Checkbox',
+  ChoiceGroup: 'Radio',
+  ComboBox: 'Dropdown',
+  ContextualMenu: 'Menu',
+  DefaultButton: 'Button',
+  DetailsList: 'DataGrid',
+  Dialog: 'Dialog',
+  DocumentCard: 'Card',
+  Dropdown: 'Dropdown',
+  Facepile: 'AvatarGroup',
+  GroupedList: 'Tree',
+  IconButton: 'Button',
+  Image: 'Image',
+  Label: 'Label',
+  Layer: 'Portal',
+  Link: 'Link',
+  MessageBar: 'MessageBar',
+  Modal: 'Dialog',
+  OverflowSet: 'Overflow',
+  Overlay: 'Portal',
+  Panel: 'Drawer',
+  Popup: 'Dialog',
+  PrimaryButton: 'Button',
+  Persona: 'Avatar',
+  Pivot: 'TabList',
+  PivotItem: 'Tab',
+  ProgressIndicator: 'ProgressBar',
+  Separator: 'Divider',
+  Shimmer: 'Skeleton',
+  Slider: 'Slider',
+  SplitButton: 'SplitButton',
+  SpinButton: 'SpinButton',
+  Spinner: 'Spinner',
+  Text: 'Text',
+  TextField: 'Input',
+  ToggleButton: 'ToggleButton',
+  Toggle: 'Switch',
+  Tooltip: 'Tooltip',
+  TooltipHost: 'Tooltip',
+  VerticalDivider: 'Divider',
+} satisfies Record<string, Migration>;
+
+/**
+ * Compatibility migrations for certain components.
+ * @see https://react.fluentui.dev/?path=/docs/concepts-migration-from-v8-component-mapping--page
+ */
+const COMPAT_MIGRATIONS = {
+  DatePicker: { component: 'DatePicker', package: '@fluentui/react-datepicker-compat' },
+  TimePicker: { component: 'TeachingPopover', package: '@fluentui/react-timepicker-compat' },
+} satisfies Record<string, Migration>;
+
+/**
+ * Preview migrations for certain components.
+ * @see https://react.fluentui.dev/?path=/docs/concepts-migration-from-v8-component-mapping--page
+ */
+const PREVIEW_MIGRATIONS = {
+  Rating: { component: 'Rating', package: '@fluentui/react-rating-preview' },
```

Comment:
This needs to be updated, these components aren't in preview anymore

Reply:
I'm also not sure if this is worth keeping in the official rule, preview components change more frequently and it might be easier to just worry about stable components in this rule.

Reply:
Updated component mapping according the info on the docsite and removed preview components for now